[Edward Alvin] ip

ThePrevailingOne opened the pull request.

Commit:
Add Gradle support

ThePrevailingOne committed:
Create an echo() and replyBye() method for Duke

ThePrevailingOne committed:
Add welcome message for Duke

ThePrevailingOne committed:
Implement word list into Duke

ThePrevailingOne committed:
Implement marking/unmarking of word list in Duke

ThePrevailingOne committed:
Refactor word list into its own class

ThePrevailingOne committed:
Create class Todo, Deadline, and Event

ThePrevailingOne committed:
Refactor code using InputType enum and an InputParser

ThePrevailingOne committed:
Refactor Duke with processInput() and move sout() statements to WordList

ThePrevailingOne committed:
Hotfix minor bug in InputParser

ThePrevailingOne committed:
Refactor and adjust WordList as well as Duke to adjust with Todo, Deadline, and Event

ThePrevailingOne committed:
Merge pull request #1 from ThePrevailingOne/skeleton

[Edward Alvin] Create Duke Skeleton, ip

ThePrevailingOne committed:
Add new Exceptions for basic invalid inputs

ThePrevailingOne committed:
Merge pull request #2 from ThePrevailingOne/exception-handling

Add new Exceptions for basic invalid inputs

ThePrevailingOne committed:
Merge branch 'master' of github.com:ThePrevailingOne/ip

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,36 @@
+        replyBye();
+    }
+
+    public static void echo(String words) {
```
Perhaps a more descriptive method name like `printLines` would be clearer!

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -26,54 +22,36 @@ public static void main(String[] args) {
-        return Duke.LIST.equals(word);
+    public static void processInput(InputType inputType, String value) {
+        switch(inputType) {
+            case LIST:
```
According to the codestyle, there should be no intentation for 'case' clauses in the switch block.

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -26,54 +22,36 @@ public static void main(String[] args) {
-    public static boolean isList(String word) {
-        return Duke.LIST.equals(word);
+    public static void processInput(InputType inputType, String value) {
+        switch(inputType) {
```
Minor style-related thing, but there should be a space before the bracket i.e `switch (inputType)`

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -25,7 +25,7 @@ public static void main(String[] args) {
             Object[] parseResult = InputParser.parseInput(input);
             InputType inputType = (InputType) parseResult[0];
-            String value = (String) parseResult[1];
+            String[] value = (String[]) parseResult[1];
```
Why the cast to String[] array here, instead of making `InputParser::parseInput(input)` return a String[] instead of Object[]?

geraldneo567 commented on src/main/java/InputParser.java:
```diff
@@ -1,13 +1,31 @@
-                value = "";
-                if (inputType.label.length() < input.length()) {
-                    value = input.substring(inputType.label.length() + 1);
+                if (inputType == InputType.BYE || inputType == InputType.LIST) {
```
Maybe it would be neater to separate out into boolean variables for example `isBye`, `isList`, etc to be in line with the code quality guideline

muraddurrani commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,85 @@
+            try {
+                input = sc.nextLine();
+                checkEmpty(input);
+                Object[] parseResult = InputParser.parseInput(input);
+                InputType inputType = (InputType) parseResult[0];
+                String[] value = (String[]) parseResult[1];
```
I think parseResult and value could both be renamed in plural form (results and values perhaps?) to be consistent with naming conventions. I noticed this in a few other places.

muraddurrani commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,85 @@
+        }
+    }
+
+    public static void processInput(InputType inputType, String[] value) {
```
Small thing because its just a parameter but maybe change value to values so its plural?

muraddurrani commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,12 @@
+public class Todo extends WordListItem{
```
I think there should be a space between WordListItem and {.

muraddurrani commented on src/main/java/UnknownInputException.java:
```diff
@@ -0,0 +1,5 @@
+public class UnknownInputException extends Exception{
```
I think there should be a space between Exception and {.

muraddurrani commented on src/main/java/WordList.java:
```diff
@@ -0,0 +1,67 @@
+import java.util.ArrayList;
+
+public class WordList {
+    private ArrayList<WordListItem> wordList;
```
Perhaps this could be renamed to words or something similar to be plural?

muraddurrani commented on src/main/java/WordList.java:
```diff
@@ -0,0 +1,67 @@
+        System.out.println(this);
+    }
+
+    public int length() {
```
The coding standards suggest that method names be verbs. Perhaps this could be changed to getLength or something like that.

muraddurrani reviewed (commented):
Generally I found it quite easy to read all the code. I think you just need to adjust some small things like renaming arrays and arraylists to be plural form.

ThePrevailingOne committed:
Implement saving system using org.json, Create JSONFileManager, and Refactor the rest of the code

ThePrevailingOne committed:
Implement date and time with DateTime to replace String

ThePrevailingOne committed:
Merge branch 'datetime'

ThePrevailingOne committed:
Add more OOP DukeUI, and edit .gitignore

ThePrevailingOne committed:
Merge branch 'a-moreoop'

ThePrevailingOne committed:
Refactor into packages and fix input whitespaces

ThePrevailingOne committed:
Merge branch 'a-packages'

ThePrevailingOne committed:
Add JUnit Testing

ThePrevailingOne committed:
Add Javadoc into the app

ThePrevailingOne committed:
Implement a find feature to Duke

ThePrevailingOne committed:
merge to add-gradle-support

ThePrevailingOne committed:
Merge branch 'master' into add-gradle-support

ThePrevailingOne committed:
Add gradle support

ThePrevailingOne committed:
Merge branch 'add-gradle-support'

ThePrevailingOne committed:
Update gradle to 7.3 and remove save file

ThePrevailingOne committed:
Add JavaFX to dependencies

ThePrevailingOne committed:
Add GUI

ThePrevailingOne committed:
Merge branch 'JavaFx'

ThePrevailingOne committed:
Hotfix data/ folder and remove unused code

ThePrevailingOne committed:
Allow user to sort the list

The sort command does not exist yet.

The sort command allows more functionality for the user which will come
really handy.

Let's,
* create the sort, sortevent, and sortdeadline command in inputtype
* update the inputparser and wordlist to adjust

ThePrevailingOne committed:
Hotfix souts

ThePrevailingOne committed:
Add a user guide

ThePrevailingOne committed:
Set theme jekyll-theme-cayman

ThePrevailingOne committed:
Set theme jekyll-theme-slate

ThePrevailingOne committed:
Set theme jekyll-theme-cayman

ThePrevailingOne commented:
# Duked
> “Your mind is for having ideas, not holding them.” – David Allen


Duked frees your mind of having to remember things you need to do. It's,
-   text-based
-   easy to learn
-   ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1.  download it from [here](https://nus-cs2103-ay2122s2.github.io/website/admin/ip-w4.html).
2.  double-click it.
3.  add your tasks.
4.  let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x]    Managing tasks
- [x]   Managing deadlines (coming soon)
- [x]   Sorting tasks (coming soon)

Just double click on the jar and it should open the app GUI :)

or

Run 
```bash
java -jar Duke.jar //depending on the relative location of the app
```

ThePrevailingOne committed:
Change user guide title

ThePrevailingOne committed:
Merge branch 'master' of github.com:ThePrevailingOne/ip

ThePrevailingOne committed:
Change gradle version
